kdtree/include/pcl/kdtree/impl/kdtree_flann.hpp

```diff
@@ -151,6 +151,14 @@ pcl::KdTreeFLANN<PointT, Dist>::nearestKSearch (const PointT &point, int k,
                            k_indices_mat, k_distances_mat,
                            k, param_k_);
 
+  const size_t& max_index = identity_mapping_ ? total_nr_points_ : index_mapping_.size();
+  k_indices.erase(std::remove_if(k_indices.begin(), k_indices.end(),
```

Comment:
Please separate this into 2 parts: `remove_if` and `erase` (even though erase-remove idiom exists) since there's a few things going on here.

Reply:
Done